src/Xamarin.Android.Build.Tasks/Tasks/GenerateResourceCaseMap.cs

```diff
@@ -0,0 +1,118 @@
+// Copyright (C) 2021 Microsoft, Inc. All rights reserved.
+using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Microsoft.Android.Build.Tasks;
+
+namespace Xamarin.Android.Tasks
+{
+	public class GenerateResourceCaseMap : AndroidTask
+	{
+		public override string TaskPrefix => "GRCM";
+		public ITaskItem[] Resources { get; set; }
+
+		public string ResourceDirectory { get; set; }
+
+		public string ProjectDir { get; set; }
+
+		public ITaskItem[] AdditionalResourceDirectories { get; set; }
+
+		public ITaskItem OutputFile { get; set; }
+
+		private Dictionary<string, string> resource_fixup = new Dictionary<string, string> (StringComparer.OrdinalIgnoreCase);
+
+		public override bool RunTask ()
+		{
+			// ResourceDirectory may be a relative path, and
+			// we need to compare it to absolute paths
+			ResourceDirectory = Path.GetFullPath (ResourceDirectory);
+
+			// Create our capitalization maps so we can support mixed case resources
+			foreach (var item in Resources) {
+				var path = Path.GetFullPath (item.ItemSpec);
+				if (!path.StartsWith (ResourceDirectory, StringComparison.OrdinalIgnoreCase)) {
+					Log.LogDebugMessage ($"Skipping {item}. Path is not include the '{ResourceDirectory}'");
+					continue;
+				}
+
+				var name = path.Substring (ResourceDirectory.Length).TrimStart ('/', '\\');
+				var logical_name = item.GetMetadata ("LogicalName").Replace ('\\', '/');
+				if (string.IsNullOrEmpty (logical_name))
+					logical_name = Path.GetFileName (path);
+
+				AddRename (name.Replace ('/', Path.DirectorySeparatorChar), logical_name.Replace ('/', Path.DirectorySeparatorChar));
+			}
+			if (AdditionalResourceDirectories != null) {
+				foreach (var additionalDir in AdditionalResourceDirectories) {
+					var dir = Path.Combine (ProjectDir, Path.GetDirectoryName (additionalDir.ItemSpec));
+					var file = Path.Combine (dir, "__res_name_case_map.txt");
+					if (!File.Exists (file)) {
+						// .NET 6 .aar files place the file in a sub-directory
+						file = Path.Combine (dir, ".net", "__res_name_case_map.txt");
+						if (!File.Exists (file))
+							continue;
+					}
+					foreach (var line in File.ReadAllLines (file).Where (l => !string.IsNullOrEmpty (l))) {
+						string [] tok = line.Split (';');
```

Comment:
It feels like this could do a loop reading line by line instead of `File.ReadAllLines()` into a big `string[]`?